Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -647,6 +647,30 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     }
     return .visitChildren
   }
+  
+  public override func visit(_ node: IfConfigDeclSyntax) -> SyntaxVisitorContinueKind {
+    for clause in node.clauses where clause.hasError {
+      if let unexpected = clause.unexpectedBetweenPoundKeywordAndCondition,
+         clause.poundKeyword.tokenKind == .poundElseifKeyword,
+         clause.poundKeyword.presence == .missing {
```

Comment:
With my comment above, I think you should also check that `unexpected` contains `#else` and `if` as its last two elements to make sure we really have the `#else if` case and didn’t randomly recover to `#elseif` by e.g. skipping an unexpected identifier.

Reply:
Here's how I'm doing it (if there's a more succinct and readable way, I'm all ears):
```swift
let unexpectedTokens = unexpectedBeforePoundKeyword
  .flatMap { $0.tokens(viewMode: .sourceAccurate) }
let lastTwoTokenKinds = unexpectedTokens
  .suffix(2)
  .map(\.tokenKind)
        
if lastTwoTokenKinds == [.poundElseKeyword, .keyword(.if)] {
  var fixedClause = clause
  fixedClause.poundKeyword = .poundElseifKeyword()
          
  addDiagnostic(
    unexpectedBeforePoundKeyword,
    StaticParserError.unexpectedPoundElseSpaceIf,
    fixIts: [
      FixIt(
        message: ReplaceTokensFixIt(replaceTokens: unexpectedTokens, replacement: fixedClause.poundKeyword),
        changes: [.replace(oldNode: Syntax(clause), newNode: Syntax(fixedClause))]
      )
    ]
  )
}
```

Reply:
You can shorten the first lines a little bit by doing the following but other than that, it’s the same way I would do it.

```swift
let unexpected = unexpectedBeforePoundKeyword.suffix(2).map { $0.as(TokenSyntax.self)?.tokenKind }
```

---

In the Fix-It you are currently replacing the entire node, which won’t render nicely in editors (e.g. the command line output will underline the entire `#else if` including the condition and suggest to replace it while keeping the condition. Instead, you should make the `poundKeyword` present and the unexpected nodes missing like the following.

```swift
[
  .makeMissing(unexpectedBeforePoundKeyword),
  .makePresent(clause.poundKeyword)
]
```

Reply:
Thank you for the suggestion. I've just pushed the changes.

Reply:
Great. That’s exactly the right solution.